### PR TITLE
Difficulty level affects bombs & Bomb population

### LINKED ISSUE #3: Bomb population & difficulty levels
Populate grid with bombs when generating it, and allow user to decide number of bombs from menu screen (?)

### PR DESCRIPTION
**Pull request by pbogre**
closes #3

also includes some minor class refactoring for readability, functionality

### CODE REVIEW

File: Minesweeper.java [hunk 1]
```diff
@@ -10,37 +10,41 @@ public class Minesweeper {
...
-    private final ActionListener actionListener = actionEvent -> {
-        Object source = actionEvent.getSource();
-    };
 
     private class Cell extends JButton {
         private final int row;
         private final int col;
-        private int value;
+        private int neighboringBombs;
+        private boolean isBomb;
```
Comment (Azgorn) [inline Minesweeper.java:18]: Why not just have a value for the cell which is either 1-8 if it's clear or a 9 if it is a bomb, thus we dont need 2 variables

File: Minesweeper.java [hunk 2]
```diff
@@ -119,9 +159,11 @@ private Menu()
 
             JPanel difficultyPanel = new JPanel();
 
-            JSlider difficultySlider = new JSlider(JSlider.HORIZONTAL, 5, 35, gridSize);
+            JSlider gridSizeSlider = new JSlider(JSlider.HORIZONTAL, 5, 35, this.selectedGridSize);
+            JSlider bombAmountSlider = new JSlider(JSlider.HORIZONTAL, 0, 99, this.selectedBombAmount);
```
Comment (Azgorn) [inline Minesweeper.java:163]: Do we want the user to select the amount of bombs manually? Shouldn't this be linked to the difficulty and be abstracted from the user?

File: Minesweeper.java (hunk 1 above)
Comment (pbogre) [reply Minesweeper.java:18]: that works perfectly fine, my change is simply for readability's sake (e.g. you dont have to remember that `value=10` means its a bomb)

File: Minesweeper.java (hunk 2 above)
Comment (pbogre) [reply Minesweeper.java:163]: im not sure what you mean by "abstracted from the user". The amount of bombs is automatically generated if you choose one of the preset difficulty levels, whereas if the user selects a custom difficulty they can choose a custom grid size as well as a custom amount of bombs to go with it. I think it's more complete this way; allows the user full control when choosing a custom difficulty.

Comment (Azgorn) [reply Minesweeper.java:163]: ah yeah mb, i thought the slider was accesible without making the difficulty custom

**Review by Azgorn — APPROVED** [review]

**Pull request merged**